packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2181,11 +2181,18 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
   }
 
   /// Replace composing region with specified text.
-  void replaceComposingRegion(SelectionChangedCause cause, String text) {
+  void replaceComposingRegion({required SelectionChangedCause cause, required String text, TextRange? composingRegionRange, bool shouldSelectWordEdgeAfterReplacement = false}) {
```

Comment:
Particularly looking for feedback here. Need a way to do this same logic without access to the composing region for iOS, and this is what I came up with. Also considered a separate method for that case.

Reply:
Check out [ReplaceTextIntent](https://master-api.flutter.dev/flutter/widgets/ReplaceTextIntent-class.html).  I think it basically does the same thing as this and the API is nearly identical, so I think you're on the right track.  I would just make these parameters match as closely with those as possible in name and order.

I wonder if this could be somehow combined with _replaceText, which uses ReplaceTextIntent in this file?

Also, maybe Android should be using this full API anyway, instead of relying on the composing region here.  Especially since non-Gboard IMEs might not have a composing region, either.  Maybe that's not this PR's responsibility, though.

Then I would drop the "composing region" terminology from this method.

This could be a breaking change so keep an eye out, but it's probably very unlikely since this was just released.

Reply:
We will ultimately need to move away from using the composing region to fix the Samsung and iOS issue (https://github.com/flutter/flutter/issues/119534), so I took your advice on not using it for Android or iOS. This method uses _replaceText already, which is used in other ways, as well, so I'm not sure if we should combine it. Maybe I could name this something like `replaceTextFromToolbar` since it contains toolbar-specific behavior.

Reply:
I talked to @LongCatIsLooong and @Renzo-Olivares about this.  We came to some good conclusions about how to organize the code that updates TextEditingValue in EditableText, but I'm worried they won't be possible here...

We'd like to keep all mutations of TextEditingValue like this in userUpdateTextEditingValue.  So in this case that would mean deleting replaceComposingRegion/replaceText and putting all of that logic inside of SpellCheckSuggestionsToolbar.  The one reason I'm not sure if that will work is the call to _didChangeTextEditingValue, which is private...

Can you try this out and see if you can get it to work without being able to call _didChangeTextEditingValue directly?  If not we're probably ok to turn this into a public `replaceWithSuggestion` method or something like that.  Currently the other toolbar buttons like cut/copy/paste all have their own public methods on EditableTextState, so it wouldn't be too terrible if this has a public method too.